examples/nlp/pretraining_BERT.py

```diff
@@ -0,0 +1,481 @@
+either actually follows A, or B is randomly sampled from somewhere else in the corpus.
+It should also generate a corresponding label for each pair, which is 1 if B actually
+follows A and 0 if not.
+- Tokenize the text dataset into it's corresponding token ids that will be used for
+embedding look-up in BERT
+- Create additional inputs for the model like `token_type_ids`, `attention_mask`, etc.
+"""
+
+# We define the maximum number of tokens after tokenization that each training sample
+# will have
+max_num_tokens = BLOCK_SIZE - tokenizer.num_special_tokens_to_add(pair=True)
+
+
+def prepare_train_features(examples):
+
+    """
+    Input:
+    {"text":Raw Documents
+    (eg: ["I love Keras. I love Tensorflow"])}
+    Output:
+    {"input_ids": Tokenized, Concatnated, and Batched Sentences
+    from the individual raw documents (batch_size x MAX_LENGTH)
+    (eg: [2, 45, 3786, 12242, 1513, 18, 45, 3786, 10667, 1691, 1021, 23473, 3])
+    "token_type_ids" : (batch_size x MAX_LENGTH) ∈ {1,0}
+    (0 for senetence 1, 1 for sentence 2, 0 for padding)
+    "attention_mask": (batch_size x MAX_LENGTH) ∈ {1,0}
+    (1 for non padded tokens, 0 for padded)
+    "next_sentence_label": (batch_size) ∈ {1,0}
+    (1 if the second sentence actually folloows the first, 0 if the senetence
+    is sampled from somewhere else in the corpus)
+    }
+    """
+
+    # Remove un-wanted samples from the training set
+    examples["document"] = [
+        d.strip() for d in examples["text"] if len(d) > 0 and not d.startswith(" =")
+    ]
+    # Split the documents from the dataset into it's individual sentences
+    examples["sentences"] = [
+        nltk.tokenize.sent_tokenize(document) for document in examples["document"]
+    ]
+    # Convert the tokens into ids using the trained tokenizer
+    examples["tokenized_sentences"] = [
+        [tokenizer.convert_tokens_to_ids(tokenizer.tokenize(sent)) for sent in doc]
+        for doc in examples["sentences"]
+    ]
+
+    # Define the outputs
+    examples["input_ids"] = []
+    examples["token_type_ids"] = []
+    examples["attention_mask"] = []
+    examples["next_sentence_label"] = []
+
+    for doc_index, document in enumerate(examples["tokenized_sentences"]):
+
+        current_chunk = []  # a buffer stored current working segments
+
+        current_length = 0
+        i = 0
+
+        # We *usually* want to fill up the entire sequence since we are padding
+        # to `block_size` anyways, so short sequences are generally wasted
+        # computation. However, we *sometimes*
+        # (i.e., short_seq_prob == 0.1 == 10% of the time) want to use shorter
+        # sequences to minimize the mismatch between pretraining and fine-tuning.
+        # The `target_seq_length` is just a rough target however, whereas
+        # `block_size` is a hard limit.
+        target_seq_length = max_num_tokens
+
+        if random.random() < SHORT_SEQ_PROB:
```

Comment:
Since you use randomness, please also seed your code via `keras.utils.set_random_seed`

Reply:
Added this!